src/plugins/intel_cpu/src/emitters/jit_snippets_emitters.cpp

```diff
@@ -123,266 +205,251 @@ void KernelEmitter::validate_arguments(const std::vector<size_t> &in,
                                        const std::vector<size_t> &out,
                                        const std::vector<size_t> &pool,
                                        const std::vector<size_t> &gpr) const {
-    if (in.size() != 2)
-        IE_THROW() << "KernelEmitter got invalid number of inputs. Expected 2, got " << in.size();
+    if (!in.empty())
+        IE_THROW() << "KernelEmitter got invalid number of inputs. Expected 0, got " << in.size();
     if (!out.empty())
-        IE_THROW() << "KKernelEmitter got invalid number of outputs. Expected 0, got " << out.size();
+        IE_THROW() << "KernelEmitter got invalid number of outputs. Expected 0, got " << out.size();
+    const auto num_params = num_inputs + num_outputs;
+    // The number of used gpr may be >= num_params since LoopBegin+LoopEnd could also use gpr to store work_amount
+    if (data_ptr_regs_idx.size() != num_params)
+        IE_THROW() << "KernelEmitter arguments are inconsistent with the gpr_regs_used size: in[0] + in[1] = "
```

Comment:
Seems like that we should change exception message now after refactoring :)
Because at the moment count of parameters and results we get inside `KernelEmitter` ctor (not parameters)
What do you think?